### DISCUSSION TITLE
Playing different music depending on int value error

### DISCUSSION THREAD

**Discussion by NikRetaNCAM** [HaxeFlixel/flixel#2485] [Q&A] [answered]
So i want do make an automatic system that will define which sound to play depending on int "curSong" value:
![image](https://user-images.githubusercontent.com/60806448/150525070-86eaa8d4-688d-4fb2-8da9-272e5f61b029.png)
yet each time i TRY to call this function (it is assigned to a button) it's silence
YET code that uses pretty much the same system:
![image](https://user-images.githubusercontent.com/60806448/150525229-5a1ab23f-4d24-433d-8087-74618dae4ddf.png)
works like a charm!

what am i doing wrong?
(also i forgot to mention that files are located like: assets/music/1/song.mp3. 1 should be able to be replaced with any other int value like 2, 25, etc)

**Comment by NikRetaNCAM** [comment, marked as answer]
ok NOW music does play at least if i tell it manually what to play
(AssetPaths.song__ogg or string path like assets/music/1/song.ogg)

**Reply by RichardBray** [reply]
Nice, I was about to say if you use string literals you shouldn't need to convert your int to a string
```haxe
class Test {
  static function main() {
    final int = 14;
    trace('Haxe is great $int!');
  }
}
```

**Reply by NikRetaNCAM** [reply]
> Nice, I was about to say if you use string literals you shouldn't need to convert your int to a string
> 
> ```hx
> class Test {
>   static function main() {
>     final int = 14;
>     trace('Haxe is great $int!');
>   }
> }
> ```

1. what does 'final' mean
2. idk if trace thing is important, but it doesnt work for me (lime test works for me the same as lime build for some reason)
3. so i just wanna go with "assets/music/" + curSong + "song.mp3"?
edit: wow, it doesn't work!

**Reply by NikRetaNCAM** [reply]
am i even doing it right? 
![image](https://user-images.githubusercontent.com/60806448/150540991-002feb08-7e7f-4dd0-b34b-f6094d79c443.png)
(curSong equals 1 and the path to the wanted .mp3 is "assets/music/1/song.mp3")

**Reply by NikRetaNCAM** [reply]
alr, changed the thing to 
![image](https://user-images.githubusercontent.com/60806448/150541739-7cd872e5-83ed-4486-b830-7afce145f01a.png)
and it works now -_-

**Reply by Geokureli** [reply]
the only difference I see is that your original path used a .mp3, and the new one used a .ogg file extension. could that be the reason? because otherwise the generated strings are identical

**Reply by RichardBray** [reply]
Yeah, I agree with @Geokureli they look exactly the same apart from the file extensions.
`final` means the variable won't be changed.

**Reply by NikRetaNCAM** [reply]
> Yeah, I agree with @Geokureli they look exactly the same apart from the file extensions. `final` means the variable won't be changed.

i know that they're same
idk why that works -_-

**Reply by Geokureli** [reply]
they are NOT the same. one is an .ogg the other is a .mp3. that's probably why
